### DISCUSSION TITLE
Quando um editor de código começa a ser uma IDE e qual a diferença

### DISCUSSION THREAD

**Discussion by willyantomaz** [TI-UNICESUMAR/2024-desafio-profissional-v-ESOFT5S-A#3] [General]
Já vi varias pessoas dizendo que o VSCode é um editor de código não uma IDE, quando fui pesquisar sobre vi que tem os editores de texto (Sublime Text, Notepad++) , editor de código(VSCode) e IDE(Intellij, AndroidStudio). 
  Uma IDE também é um editor de código ? IDE também é um editor de texto ? quando um editor de código pode ser chamado de IDE ?

**Comment by joaoagr1** [comment]
Acredito que oque define uma IDE sãos as ferramentas e funcionalidades que vão além da edição de texto básica e se estendem ao desenvolvimento de software de forma mais abrangente, incluindo depuração, gerenciamento de projetos, integração com sistemas de controle de versão, entre outros, que tornam o desenvolvimento mais fácil.

Esse artigo da alura fala bem a respeito disso:
https://www.alura.com.br/artigos/o-que-e-uma-ide#:~:text=Editores%20de%20c%C3%B3digo%20s%C3%A3o%20editores,tornar%20a%20codifica%C3%A7%C3%A3o%20mais%20f%C3%A1cil.

**Comment by AndreJacobx** [comment]
acredito que um editor de texto só passa a ser uma IDE quando ele disponibiliza recursos focados no desenvolvimento de software, por exemplo, nós conseguimos escrever código em qualquer lugar que possua um editor de texto. Agora executar esse código e trabalhar em cima dele com os recursos da linguagem usada é uma tarefa que só as IDEs suportam
